src/Sylius/Bundle/AdminBundle/Twig/Component/Shared/Navbar/NotificationsComponent.php

```diff
@@ -13,16 +13,89 @@
 
 namespace Sylius\Bundle\AdminBundle\Twig\Component\Shared\Navbar;
 
+use Psr\Clock\ClockInterface;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Sylius\Bundle\CoreBundle\SyliusCoreBundle;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
 use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
 
 class NotificationsComponent
 {
-    /**
-     * @return array<string, mixed>
-     */
+    public const LATEST_SYLIUS_VERSION_CACHE_KEY = 'latest_sylius_version';
+
+    public function __construct(
+        private readonly ClientInterface $client,
+        private readonly RequestStack $requestStack,
+        private readonly RequestFactoryInterface $requestFactory,
+        private readonly StreamFactoryInterface $streamFactory,
+        private readonly CacheInterface $cache,
+        private readonly ClockInterface $clock,
+        private readonly string $hubUri,
+        private readonly string $environment,
+    ) {
+    }
+
+    /** @return array<array-key, mixed> */
     #[ExposeInTemplate(name: 'notifications')]
     public function getNotifications(): array
```

Comment:
Given what it's called and also how the template is structured I'd be for introducing a composite inside that'll allow anyone to insert their own notifications